### PR TITLE
Add elementwise multiplication

### PR DESCRIPTION
**Pull request by sashaDoubov**
Also adds a new eltwise_ops file + accompanying test.

### CODE REVIEW

File: stk/ops/eltwise_ops.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from stk.matrix import Matrix
+
+def mul(a, b):
+    """
+    Element-wise multiplication of matrices a and b.
```
Comment (tgale96) [inline stk/ops/eltwise_ops.py:5]: The formatting on this docstring is a little funky - I try to use Google Python docstring style

https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

File: stk/ops/eltwise_ops_test.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    dense_data = torch.randn_like(x.data) * std
+    sparse = stk.Matrix(x.size(),
+                        dense_data,
+                        x.row_indices,
+                        x.column_indices,
+                        x.offsets)
+    dense = stk.ops.to_dense(sparse)
+
+    cuda_device = torch.device("cuda")
+    return (dense.to(cuda_device).requires_grad_(True),
```
Comment (tgale96) [inline stk/ops/eltwise_ops_test.py:48]: I think what you want here is for the output device to match the input device? You should be able to get that by passing `device=x.device` to the `randn_like` call. After that, all the other tensors should be on that device.

File: stk/ops/eltwise_ops_test.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        self.assertEqual(grad.dim(), 2)
+        self.assertEqual(expected_grad.size()[0], grad.size()[0])
+        self.assertEqual(expected_grad.size()[1], grad.size()[1])
+        self.assertTrue(allclose(grad, expected_grad))
+
+        # RHS gradient.
+        grad =  stk.ops.to_dense(b.grad)
+        expected_grad = b_dense.grad
+        self.assertEqual(grad.dim(), 2)
+        self.assertEqual(expected_grad.size()[0], grad.size()[0])
```
Comment (tgale96) [inline stk/ops/eltwise_ops_test.py:85]: I think we can check shape equality with just `==`?

File: stk/ops/eltwise_ops.py (hunk 1 above)
Comment (sashaDoubov) [reply stk/ops/eltwise_ops.py]: updated, please lmk if that looks better.

File: stk/ops/eltwise_ops_test.py (hunk 3 above)
Comment (sashaDoubov) [reply stk/ops/eltwise_ops_test.py]: I kept the `assertEqual` but got rid of the individual checks for `.size()[0]` and `.size()[1]`, is that okay?

File: stk/ops/eltwise_ops.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+def mul(a, b):
+    """Performs element-wise multiplication of matrices a and b.
+
+    It is the user's responsibility to make sure that a and b
+    follow the same matrix topology. This function assumes it is safe
+    to use the topoplogy of a.
+
+    Args:
+        a:
+           sparse stk.matrix.Matrix.
```
Comment (tgale96) [inline stk/ops/eltwise_ops.py:12]: You can put the description of each arg following the colon

`a: stk.Matrix`

File: stk/ops/eltwise_ops.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        b:
+           sparse stk.matrix.Matrix with a's matrix topology.
```
Comment (tgale96) [inline stk/ops/eltwise_ops.py:14]: The public type here is actually `stk.Matrix` - we expose that API in the `__init__.py` file.

**Review by tgale96 — APPROVED** [review]

File: stk/ops/eltwise_ops.py (hunk 5 above)
Comment (sashaDoubov) [reply stk/ops/eltwise_ops.py]: nice, fixed!

File: stk/ops/eltwise_ops.py (hunk 4 above)
Comment (sashaDoubov) [reply stk/ops/eltwise_ops.py]: fixed!

**Comment by sashaDoubov** [comment]
I've updated the docstrings. I don't have write access to the repo, but if you can merge if it looks good that would be great!

**Comment by tgale96** [comment]
Thanks, Sasha!

**Pull request merged**